src/Service/DogCRUDService.php

```diff
@@ -37,12 +37,44 @@ public function savePostForDog(Dog $dog,Post $post): void
      * @accepts Dog $dog
      * @return Post[]
      */
-    public function getAllPosts(Dog $dog): iterable
+    public function getAllPosts(Dog $dog): array
     {
         return $this->postRepository->findAllPostsByDog($dog);
     }
+    
+    /**
+    * @return Dog[]
+    */
+    public function getAllDogs(): array{
+        return $this->dogRepository->findAll();
+    }
 
-    public function getDogNiceName(Dog $dogUser): string {
+    public function getDogNiceName(Dog $dogUser): string
+    {
         return ucwords($dogUser->getUserIdentifier());
     }
+
+    /**
+     * @paramter Dog[] $dogs
+     * @paramter Dog[] $myPack
+     * @return Dog[]
+     */
+    public function getDogsNotInPack(array $myPack, array $dogs): array
+    {
+        if(count($myPack) === 0 || count($dogs) === 0) {
+            return [];
+        }
+
+        $myPack = array_map(fn(Dog $dog): int => $dog->getId(), $myPack);
+
+        $otherDogUsers = [];
+
+        foreach ( $dogs as $dog) {
+            if (!in_array($dog->getId(),$myPack)) {
+                array_push($otherDogUsers,$dog);
+            }
+        }
+
+        return $otherDogUsers;
+    }
```

Comment:
I feel like this lives in the PackService. Dogs know of their friendships via this in my head. Not sure how this fits into the CRUD responsibility the DogCRUDService owns.

Reply:
I've moved this function to the packService. Your comment makes sense.